guacamole-ext/src/main/java/org/apache/guacamole/properties/IPAddressListProperty.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.guacamole.properties;
+
+import inet.ipaddr.IPAddress;
+import inet.ipaddr.IPAddressString;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+import org.apache.guacamole.GuacamoleException;
+import org.apache.guacamole.GuacamoleServerException;
+
+/**
+ * A GuacamoleProperty implementation that parses a String for a comma-separated
+ * list of IP addresses and/or IP subnets, both IPv4 and IPv6, and returns the
+ * list of those valid IP addresses/subnets.
+ */
+public abstract class IPAddressListProperty implements GuacamoleProperty<List <IPAddress>> {
+    
+    /**
+     * A pattern which matches against the delimiters between values. This is
+     * currently simply a comma and any following whitespace. Parts of the
+     * input string which match this pattern will not be included in the parsed
+     * result.
+     */
+    private static final Pattern DELIMITER_PATTERN = Pattern.compile(",\\s*");
+    
+    @Override
+    public List<IPAddress> parseValue(String values) throws GuacamoleException {
```

Comment:
I'm on the fence about directly exposing an object from a third-party library as part of the extension API. Should we ever decide to switch to a different IP address library, this would encumber that switch.

Would Java's standard `InetAddress` work? I see `IPAddress` provides a `toInetAddress()`.

Reply:
I think the main challenge with that approach is that the list, as implemented with the `IPAddress` class, is designed to take a combination of IPv4 addresses (1.2.3.4), IPv4 CIDR notation (1.2.3.0/24), IPv6 addresses, and IPv6 CIDR notation, and handle all of those possibilities in a single list. I haven't found anything conclusive, nor have I tested it, but I'm pretty sure the built-in `InetAddress` class can only handle IP addresses - possibly either IPv4 or IPv6 - but not CIDR notation.

Reply:
@mike-jumper Any further thoughts on this one?

Reply:
@mike-jumper Ping...

Reply:
Yeah, I think you're right. It'd be nice if there were a standard class, but there really isn't anything outside `IPAddress` from this library, and that library seems good.